File: 2607. Make K-Subarray Sums Equal/Solution.java
class Solution {
    public long makeSubKSumEqual(int[] arr, int k) {
        long total = 0;
        int n = arr.length;
        boolean[] seen = new boolean[n];
        for (int i = 0; i < n; i++) {
            if (seen[i]) continue;
            List<Integer> a = new ArrayList<>();
            a.add(arr[i]);
            int j = (i + k) % n;
            while (j != i) {
                a.add(arr[j]);
                seen[j] = true;
                j = (j + k) % n;
            }
            Collections.sort(a);
            int median = a.get(a.size() / 2);
            for (int b = 0; b < a.size(); b++) {
                total += Math.abs(a.get(b) - median);
            }
        }
        return total;
    }
}

File: 2607. Make K-Subarray Sums Equal/Solution.py
class Solution:
    def makeSubKSumEqual(self, arr: List[int], k: int) -> int:
        if k == 1:
            a = sorted(arr)
            median = a[len(a) // 2]
            return sum([abs(a[i] - median) for i in range(len(a))])
        total, n = 0, len(arr)
        visited = [False for i in range(n)]
        for i in range(n):
            if visited[i]: continue
            a = [arr[i]]
            j = (i + k) % n
            while j != i:
                visited[j] = True
                a.append(arr[j])
                j = (j + k) % n
            a = sorted(a)
            median = a[len(a) // 2]
            total += sum([abs(a[i] - median) for i in range(len(a))])
        return total